Add docs for nack backoff policy (#12660)
apache/pulsar: site2/docs/concepts-messaging.md

> **Note**  
> If batching is enabled, all messages in one batch are redelivered to the consumer.

> ### Negative Redelivery Backoff

Reviewer: ```suggestion
### Negative redelivery backoff
```

> In general, the consumer is not able to process the message successfully, what we can use [Negative acknowledgement](concepts-messaging.md#Negative acknowledgement) and redeliver the message after processing the message failure so that the message can be redelivered to other consumers(for the Shared subscription).

Reviewer: ```suggestion
In general, consumers are not able to process messages successfully. In this case, you can use [negative acknowledgement](concepts-messaging.md#negative-acknowledgement) and redeliver the messages after processing message failures, so that the messages can be redelivered to other consumers (for the Shared subscription).
```

do you mean this?

> But this is not flexible enough, so we introduce a redelivery backoff mechanism which we can achieve redelivery with different delays according to the number of times the message is retried.

Reviewer: ```suggestion
But this is not flexible enough. A better way is to use the **redelivery backoff mechanism**.  You can redeliver messages with different delays by setting the number of times the messages is retried.
```
do you mean this?
